src/packagedcode/conan.py

```diff
@@ -0,0 +1,281 @@
+            self.homepage_url = values.value
+        elif variable_name == "url":
+            self.vcs_url = values.value
+        elif variable_name == "license":
+            self.license = values.value
+        elif variable_name == "topics":
+            self.keywords.extend(
+                [el.value for el in values.elts if isinstance(el, ast.Constant)]
+            )
+        elif variable_name == "requires":
+            if isinstance(values, ast.Tuple):
+                self.requires.extend(
+                    [el.value for el in values.elts if isinstance(el, ast.Constant)]
+                )
+            elif isinstance(values, ast.Constant):
+                self.requires.append(values.value)
+
+    def visit_Call(self, node):
+        if not isinstance(node.func, ast.Attribute) or not isinstance(
+            node.func.value, ast.Name
+        ):
+            return
+        if node.func.value.id == "self" and node.func.attr == "requires":
+            if node.args and isinstance(node.args[0], ast.Constant):
+                self.requires.append(node.args[0].value)
+
+
+class ConanFileHandler(models.DatafileHandler):
+    datasource_id = "conan_conanfile_py"
+    path_patterns = ("*/conanfile.py",)
+    default_package_type = "conan"
+    default_primary_language = "C++"
+    description = "conan recipe"
+    documentation_url = "https://docs.conan.io/2.0/reference/conanfile.html"
+
+    @classmethod
+    def parse(cls, location):
+        with io.open(location, encoding="utf-8") as loc:
+            conan_recipe = loc.read()
+
+        try:
+            tree = ast.parse(conan_recipe)
+            recipe_class_def = next(
+                (
+                    node
+                    for node in tree.body
+                    if isinstance(node, ast.ClassDef)
+                    and node.bases
+                    and isinstance(node.bases[0], ast.Name)
+                    and node.bases[0].id == "ConanFile"
+                ),
+                None,
+            )
+
+            parser = ConanFileParser()
+            parser.visit(recipe_class_def)
+        except SyntaxError as e:
+            if TRACE:
+                logger_debug(f"Syntax error in conan recipe: {e}")
+            return
+
+        if TRACE:
+            logger_debug(f"ConanFileHandler: parse: package: {parser.to_dict()}")
+
+        dependencies = get_dependencies(parser.requires)
+
+        yield models.PackageData(
+            datasource_id=cls.datasource_id,
+            type=cls.default_package_type,
+            primary_language=cls.default_primary_language,
+            namespace=None,
+            name=parser.name,
+            version=parser.version,
+            description=parser.description,
+            homepage_url=parser.homepage_url,
+            keywords=parser.keywords,
+            extracted_license_statement=parser.license,
+            dependencies=dependencies,
+        )
+
+
+class ConanDataHandler(models.DatafileHandler):
```

Comment:
I saw some different kinds of conandata.yml too like this  https://github.com/Ultimaker/Cura/blob/main/conandata.yml is this seen often? do we support this? Is this a different version of conandata?

Reply:
Technically speaking `conandata.yml` can contain arbitrary information, but we're only interested in the part specified here https://docs.conan.io/2/tutorial/creating_packages/handle_sources_in_packages.html#using-the-conandata-yml-file. This section of conandata.yml specifies how to provide the url and sha256 for each version.
Apart from that, developers can specify any other information in this YML file that they might need to automate the `conanfile.py`.